runtime/compiler/env/VMJ9.cpp

```diff
@@ -1865,6 +1865,24 @@ UDATA TR_J9VMBase::thisThreadGetOSRReturnAddressOffset()
    return offsetof(J9VMThread, osrReturnAddress);
    }
 
+#if defined(TR_TARGET_S390)
+/**
+ * @brief TDB is Transaction Diagnostic Block used for Transactional Memory Debugging.
+ * It is a 256 Byte block stored in the J9VMThread and must be 8 byte boundary aligned.
+ */
+uint16_t TR_J9VMBase::thisThreadGetTDBOffset()
+   {
+#if !defined(J9ZTPF)
+   // This assume is valid to confirm the TDB is on an 8 byte boundary because we know the VMThread is already aligned on a 256 byte boundary
+   // If this assume fails, then that means someone modified the J9VMThread structure which has thrown off the alignment.
+   TR_ASSERT(offsetof(J9VMThread, transactionDiagBlock)%8 == 0, "The Transaction Diagnostic Block must be aligned on a doubleword i.e. 8 byte boundary");
+   return offsetof(J9VMThread, transactionDiagBlock);
+#else
+   return 0;
```

Comment:
Is `0` the right value for the case where there is no `transactionDiagBlock`?  Won't that cause the vmThread to be corrupted if this code was incorrectly run on any other platform?

In VM code, we'd use a TraceAssert to mark this case as illegal.

Reply:
That's from `#if !defined(J9ZTPF)`. I can't comment on that. I didn't add that `ifdef`, the zptf people did. It's been like that for a long time on older pre-open source versions with no issues, so I assume the code works for them.

Reply:
Likely on the assumption that it's never called.  Better to be explicit than to leave potential corruption points in the code.

Reply:
Can you suggest how I should fix this? I don't follow what you mean by "use a TraceAssert to mark this case as illegal".

Reply:
Similar to the trace assert used elsewhere in the JIT.  ie https://github.com/eclipse/openj9/blob/afbe2b92c85fe01ef44df0b6b5ce45421b241f5f/runtime/compiler/control/CompilationThread.cpp#L5265

Reply:
Will that assert fire in a prod/released build?

Reply:
Yes.  Similar to all asserts in the VM / GC code.

Reply:
Upon further reflection, I need to discuss with a ztpf dev about what should be done here. Switching the `return 0` to a fatal assert seems risky without consulting them.

Reply:
Ok. Waiting on some testing, then will remove WIP.